inductiva/resources/machines_base.py

```diff
@@ -43,13 +45,15 @@ def __init__(self,
                 example, when retrieving with the `machines_groups.get` method.
                 Users should not set this argument in anyway.
         """
-        if machine_type not in inductiva.resources.list_available_machines():
-            raise ValueError("Machine type not supported")
+        if machine_type not in inductiva.resources.list_available_machines(
+                provider.lower()):
+            raise ValueError(f"Machine type not supported in {provider}")
```

Comment:
just to prevent the weird line break:

```suggestion
        provider = provider.lower()
        available = inductiva.resources.list_available_machines(provider)
        if machine_type not in available:
            raise ValueError(f"Machine type not supported in {provider}")
```